kaldb/src/main/java/com/slack/kaldb/writer/JsonLogFormatter.java

```diff
@@ -0,0 +1,55 @@
+package com.slack.kaldb.writer;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.slack.kaldb.logstore.LogMessage;
+import com.slack.kaldb.util.JsonUtil;
+import com.slack.service.murron.trace.Trace;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+public class JsonLogFormatter {
```

Comment:
Add a comment here explaining what this class is for, and who is expected to use it. Specifically I can see how calling out the random UUIDs would be important, and how the default values are handled when they do not exist.